modules/github_integration/lib/api/v3/github_integration/github_pull_request_representer.rb

```diff
@@ -0,0 +1,166 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+require 'roar/decorator'
+require 'roar/json/hal'
+
+module API
+  module V3
+    module GithubIntegration
+      class GithubPullRequestRepresenter < ::API::Decorators::Single
+        include API::Caching::CachedRepresenter
+        include API::Decorators::DateProperty
+        include API::Decorators::LinkedResource
+
+        cached_representer key_parts: %i[github_user merged_by latest_check_runs],
+                           disabled: false
+
+        link :staticPath do
+          next
+        end
+
+        property :id
+
+        property :number,
+                 setter: ->(*) { nil }
```

Comment:
Why do you consider the `setter: ->(*) { nil }` statements to be necessary?

Reply:
I saw a similar pattern for `getter` in other representers that seemed to disabled api output. Since this is a read-only API i thought I'd better disable input (create, update) on the representer-level. But since was just an assumption that was wrong, I happily remove it